sdk/include/cheri.h

```diff
@@ -0,0 +1,25 @@
+// Copyright CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include <compartment-macros.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/**
+ * Checks that `ptr` is valid, unsealed, does not overlap the caller's stack,
+ * has at least `rawPermissions`, and has at least `space` bytes after the
+ * current offset.
+ *
+ * If the permissions do not include Global, then this will also check that the
+ * capability does not point to the current thread's stack.  This behaviour can
+ * be disabled (for example, for use in a shared library) by passing `false`
+ * for `checkStackNeeded`.
```

Comment:
```suggestion
 * If the permissions do not include Global and `checkStackNeeded` is `false`, then this will also check that the
  * capability does not point to the current thread's stack.
```